### PR TITLE
Support Ipywidgets 8.0 to support Python 3.10

### PR DESCRIPTION
**Pull request by aarondr77**
# Description

We've observed users installing Mito on Python 3.10+ run into the same issues over and over. 

```
$ python -m mitoinstaller install --no-cache-dir
Starting install...
Upgrade mitoinstaller
Setting up environment
Check dependencies
Remove mitosheet3 if present
Install mitosheet
Activate extension
Traceback (most recent call last):
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/mitoinstaller/installer_steps/installer_step.py", line 44, in execute
    self.execution_function()
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/mitoinstaller/installer_steps/mitosheet_installer_steps.py", line 77, in install_step_mitosheet_activate_notebook_extension
    run_command([sys.executable, "-m", 'jupyter', 'nbextension', 'install', '--py', '--user', 'mitosheet'])
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/mitoinstaller/commands.py", line 50, in run_command
    raise Exception(get_output(completed_process))
Exception: Traceback (most recent call last):
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/bin/jupyter-nbextension", line 8, in <module>
    sys.exit(main())
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/jupyter_core/application.py", line 269, in launch_instance
    return super().launch_instance(argv=argv, **kwargs)
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/traitlets/config/application.py", line 976, in launch_instance
    app.start()
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/nbclassic/nbextensions.py", line 972, in start
    super().start()
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/jupyter_core/application.py", line 258, in start
    self.subapp.start()
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/nbclassic/nbextensions.py", line 702, in start
    self.install_extensions()
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/nbclassic/nbextensions.py", line 675, in install_extensions
    full_dests = install(self.extra_args[0],
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/nbclassic/nbextensions.py", line 203, in install_nbextension_python
    m, nbexts = _get_nbextension_metadata(module)
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/nbclassic/nbextensions.py", line 1107, in _get_nbextension_metadata
    m = import_item(module)
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/traitlets/utils/importstring.py", line 38, in import_item
    return __import__(parts[0])
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/mitosheet/__init__.py", line 31, in <module>
    from mitosheet.mito_widget import MitoWidget, sheet
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/mitosheet/mito_widget.py", line 17, in <module>
    from ipywidgets import DOMWidget
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/ipywidgets/__init__.py", line 25, in <module>
    from .widgets import *
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/ipywidgets/widgets/__init__.py", line 20, in <module>
    from .widget_selection import RadioButtons, ToggleButtons, ToggleButtonsStyle, Dropdown, Select, SelectionSlider, SelectMultiple, SelectionRangeSlider
  File "/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/site-packages/ipywidgets/widgets/widget_selection.py", line 9, in <module>
    from collections import Mapping, Iterable
ImportError: cannot import name 'Mapping' from 'collections' (/Users/aarondiamond-reivich/opt/anaconda3/envs/mitoenv/lib/python3.10/collections/__init__.py)

```
This error is thrown in the ipywidgets codebase. The error is caused because in Python 3.10, instead of importing `from collections import Mapping`, you must import as `from collections.abc import Mapping`.

Ipywdigets 8.0 resolves this import issue. See [here](https://github.com/jupyter-widgets/ipywidgets/blob/3dbd281a7488da6d2e2fc10d14df58041963eaa6/python/ipywidgets/ipywidgets/widgets/widget_selection.py).

This PR updates Mito to work with ipywidgets 8.0

See investigation and documentation about this upgrade process [here](https://www.notion.so/trymito/Python-3-10-Error-cbec7bf27bae42e58083ece3bc052082)

In upgrading my Python version to 3.10, I also ran into issues with the development environmnent. Specifically:
1. `ensure_python` is deprecated on 3.10, so I switched to `python_requires`
2. removing `ensure_python` caused GitHub tests to not pass. I was able to resolve this by following the error message and updating `InitializeParameters` to `IInitializeParameters` in `src/jupyter/widget.tsx`

Fixing 2 above, caused Mito on JupyterLab 2 to not work, however. It says:
```
"src/jupyter/widget.tsx(69,39): error TS2724: '"/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@jupyter-widgets/base/lib/widget".WidgetView' has no exported member named 'IInitializeParameters'. Did you mean 'InitializeParameters'?"
```

# Testing

To test:
1. Install python 3.10
2. Checkout the main branch
4. Remove the ensure_python import and use in the setup.py file
5. Create a new virtual evironment and run the commands to start Mito (note you might need to install a xcode upgrade, just read the error message)
6. Notice that that you receive the error ` ImportError("cannot import name 'Mapping' from 'collections' `
7. Checkout this branch
8. Create a new virtual evironment and run the commands to start Mito
9. See that Mito works 

# Documentation

Note if any new documentation needs to addressed or reviewed.

### CODE REVIEW

**Comment by naterush** [comment]
Ok see the 7.0 -> 8.0 migration guide here: https://github.com/jupyter-widgets/ipywidgets/blob/master/docs/source/migration_guides.md

See the following code: `Typescript trick: If you need to support a deserializer function against both ipywidgets 7 and older and the new version 8, you can change your deserializer function to have the following signature:`

Let's try this!

**Comment by aarondr77** [comment]
```
    $ npm run build:lib && npm run build:labextension
    
    > mitosheet2@0.1.341 build:lib
    > tsc
    
    src/plugin.tsx(24,54): error TS2344: Type 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@lumino/widgets/types/widget").Widget' does not satisfy the constraint 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@jupyter-widgets/base/node_modules/@lumino/widgets/types/widget").Widget'.
      Types of property 'title' are incompatible.
        Type 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@lumino/widgets/types/title").Title<import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@lumino/widgets/types/widget").Widget>' is not assignable to type 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@jupyter-widgets/base/node_modules/@lumino/widgets/types/title").Title<import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@jupyter-widgets/base/node_modules/@lumino/widgets/types/widget").Widget>'.
          Types have separate declarations of a private property '_label'.
    src/plugin.tsx(29,38): error TS2344: Type 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@lumino/widgets/types/widget").Widget' does not satisfy the constraint 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@jupyter-widgets/base/node_modules/@lumino/widgets/types/widget").Widget'.
    src/plugin.tsx(42,22): error TS2344: Type 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@lumino/widgets/types/widget").Widget' does not satisfy the constraint 'import("/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/node_modules/@jupyter-widgets/base/node_modules/@lumino/widgets/types/widget").Widget'.
    error Command failed with exit code 1.
    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/setup.py", line 261, in <module>
        setup(**setup_args)
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 155, in setup
        return distutils.core.setup(**attrs)
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 148, in setup
        return run_commands(dist)
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 163, in run_commands
        dist.run_commands()
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 967, in run_commands
        self.run_command(cmd)
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
        cmd_obj.run()
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/jupyter_packaging/setupbase.py", line 640, in run
        [self.run_command(cmd) for cmd in cmds]
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/jupyter_packaging/setupbase.py", line 640, in <listcomp>
        [self.run_command(cmd) for cmd in cmds]
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
        cmd_obj.run()
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/jupyter_packaging/setupbase.py", line 321, in run
        c.run()
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/jupyter_packaging/setupbase.py", line 615, in run
        builder()
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/jupyter_packaging/setupbase.py", line 214, in builder
        run(npm_cmd + ['run', build_cmd], cwd=node_package)
      File "/private/var/folders/4l/hvyxwfp95vb2j1c1f54drnz00000gn/T/pip-build-env-jx2sqk_m/overlay/lib/python3.10/site-packages/jupyter_packaging/setupbase.py", line 275, in run
        return subprocess.check_call(cmd, **kwargs)
      File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 369, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['/usr/local/bin/yarn', 'run', 'build:all']' returned non-zero exit status 1.
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/venv/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/setup.py'"'"'; __file__='"'"'/Users/aarondiamond-reivich/Documents/saga/monorepo/mitosheet/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
```

**Comment by aarondr77** [comment]
If I revert the changes made https://github.com/mito-ds/monorepo/pull/455/commits/4663c82f7a778a54b92b27b9c8c018159b7331af to the `src/plugin.tsx` file, it resolves this error, but then I am still unable to run `jlpm run watch`
```
$ jlpm run watch
yarn run v1.21.1
$ npm run watch:all

> mitosheet2@0.1.341 watch:all
> run-p watch:lib watch:labextension


> mitosheet2@0.1.341 watch:lib
> tsc -w


> mitosheet2@0.1.341 watch:labextension
> jupyter labextension watch .

[11:24:04 AM] Starting compilation in watch mode...

Please supply at least one subcommand: check, disable, enable, install, link, list, uninstall, unlink, update
ERROR: "watch:labextension" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: mitosheet/src/plugin.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import { IJupyterWidgetRegistry } from '@jupyter-widgets/base';
 import { INotebookTracker, NotebookActions } from '@jupyterlab/notebook';
 import { Application, IPlugin } from 'application';
-import { Widget } from 'widgets';
+import { Widget } from "../node_modules/@lumino/widgets/types/widget"
```
Comment (naterush) [inline mitosheet/src/plugin.tsx:11]: This is def wrong!

Comment (naterush) [reply mitosheet/src/plugin.tsx:11]: These paths don't make sense when things are built. There are no node modules distributed to the user!

**Pull request merged**